Translate /docs/concepts/services-networking/ingress-controllers/ into Japanese (#21986)
kubernetes/website: content/ja/docs/concepts/services-networking/ingress-controllers.md

---
title: Ingressコントローラー
reviewers:
content_type: concept
weight: 40
---

<!-- overview -->

> Ingressリソースが動作するためには、クラスターでIngressコントローラーが実行されている必要があります。

Reviewer: ```suggestion
Ingressリソースが動作するためには、クラスタでIngressコントローラーが実行されている必要があります。
```

Reviewer 2: @bells17 クラスターはクラスターじゃないですっけ？

Reviewer: ミスなので対応不要です
クラスターで合ってます

> `kube-controller-manager`バイナリの一部として実行される他のタイプのコントローラーとは異なり、Ingressコントローラーはクラスターで自動的に起動されません。このページを使用して、クラスターに最適なIngressコントローラーの実装を選択してください。

Reviewer: ```suggestion
`kube-controller-manager`バイナリの一部として実行される他のタイプのコントローラーとは異なり、Ingressコントローラーはクラスタで自動的に起動されません。このページを使用して、クラスタに最適なIngressコントローラーの実装を選択してください。
```

Reviewer: ミスなので対応不要です
クラスターで合ってます

* [Istio](https://istio.io/)ベースのIngressコントローラー[Control Ingress Traffic](https://istio.io/docs/tasks/traffic-management/ingress/)。
* [Kong](https://konghq.com/)は、[Kong Ingress Controller for Kubernetes](https://github.com/Kong/kubernetes-ingress-controller)の[コミュニティ版](https://discuss.konghq.com/c/kubernetes)と[商用版]](https://konghq.com/kong-enterprise/)のサポートと保守を提供しています。
* [NGINX, Inc.](https://www.nginx.com/)は[NGINX Ingress Controller for Kubernetes](https://www.nginx.com/products/nginx/kubernetes-ingress-controller)のサポートと保守を提供しています。
* [Skipper](https://opensource.zalando.com/skipper/kubernetes/ingress-controller/)は、カスタムプロキシーを構築するためのライブラリーとして設計された、Kubernetes Ingressなどのユースケースを含む、サービス構成用のHTTPルーターとリバースプロキシーです。
* [Traefik](https://github.com/containous/traefik)はフル機能([Let's Encrypt](https://letsencrypt.org), secrets, http2, websocket)のIngressコントローラーで、[Containous](https://containo.us/services)による商用サポートもあります。

## 複数のIngressコントローラーの使用 {#using-multiple-ingress-controllers}

> [任意の数のIngressコントローラー](https://git.k8s.io/ingress-nginx/docs/user-guide/multiple-ingress.md#multiple-ingress-controllers)をクラスターにデプロイできます。Ingressを作成する際には、各Ingressに適切な[`ingress.class`](https://git.k8s.io/ingress-gce/docs/faq/README.md#how-do-i-run-multiple-ingress-controllers-in-the-same-cluster)をアノテーションし、クラスター内に複数のIngressコントローラーが存在する場合には、どのIngressコントローラーを使用するかを示す必要があります。

Reviewer: > Ingressを作成する際には~

の部分は

> クラスタ内に複数のIngressコントローラーが存在する場合には、

のところが文の最初に来るべき感じがします

Reviewer: あと s/クラスター/クラスタ/ をお願いします

Reviewer: https://github.com/kubernetes/website/pull/21986#discussion_r443488430
はミスなので対応不要です
クラスターで合ってます

Reviewer 2: @oke-py ping

Author: うーん、最初に来る = 文全体にかかっているというのはdisagreeですね。こうかな
```suggestion
[任意の数のIngressコントローラー](https://git.k8s.io/ingress-nginx/docs/user-guide/multiple-ingress.md#multiple-ingress-controllers)をクラスターにデプロイできます。Ingressを作成する際には、各Ingressに適切な[`ingress.class`](https://git.k8s.io/ingress-gce/docs/faq/README.md#how-do-i-run-multiple-ingress-controllers-in-the-same-cluster)をアノテーションし、クラスター内に複数のIngressコントローラーが存在する場合にどのIngressコントローラーを使用するかを示す必要があります。
```

Reviewer 2: ```suggestion
[Ingressコントローラーは、好きな数だけ](https://git.k8s.io/ingress-nginx/docs/user-guide/multiple-ingress.md#multiple-ingress-controllers))クラスターにデプロイすることができます。Ingressを作成する際には、クラスター内に複数のIngressコントローラーが存在する場合にどのIngressコントローラーを使用するかを示すために適切な[`ingress.class`](https://git.k8s.io/ingress-gce/docs/faq/README.md#how-do-i-run-multiple-ingress-controllers-in-the-same-cluster)のアノテーションを指定します。
```

日本語的にはこういうのはどうでしょう？

クラスを定義しない場合、クラウドプロバイダーはデフォルトのIngressコントローラーを使用する場合があります。

> 理想的には、すべてのIngressコントローラーはこの仕様を満たすべきですが、さまざまなIngressコントローラーはわずかに異なる動作をします。

Reviewer 3: なんかちょっと違和感？
```suggestion
理想的には、すべてのIngressコントローラーはこの仕様を満たすべきですが、いくつかのIngressコントローラーはわずかに異なる動作をします。
```
のほうがしっくりくると思います
